### PR TITLE
build: add the build command to the MAKE -- Wheels #25

### PR DESCRIPTION
**Pull request by fbraza**
This a pull request to answer @sdebruyn raised issue.

I quickly test the `.whl` on databricks and from there you can use your skeleton functions.

Cheers

### CODE REVIEW

**Comment by vikramaditya91** [comment]
Hey @fbraza . Quick question. How did you test that the .whl works on databricks?
Did it not complain on the missing dependencies? eg. typer?

**Comment by fbraza** [comment]
Hello @vikramaditya91 

No it did not complain about that. the `.whl` is just like a `zip` file that contains compiled requirements as defined in your `pyproject.toml` file. Everything should be present in the `.whl`. But I did not use your package as if I was running it from a terminal with Typer but rather I imported the three ETL functions template that you defined. And they got imported without any issue.

I don't think it is relevant to use the CLI command in the databaricks environment. Moreover Spark session is already instantiated there. So that is why I just tested if it was possible to use the functions you defined there and it worked. ^_^

But maybe @sdebruyn could add something here about his expectation with this package on Databricks, synapse ect...

**Comment by vikramaditya91** [comment]
@fbraza It is not just for it to be used as a CLI. For example, if the `pyproject.toml` says that it needs `numpy` or `boto3`, and I am using `numpy`/`boto3` for some reason in the ETL job, then it would complain that they are missing from the environment I suppose. To package the list of dependencies in the pyproject.toml, I had this `make pack_req` which packages the dependencies in a package.zip file using the Docker image.

The question is for @sdebruyn I think.
The `poetry build` commands creates a `.whl` and `.zip` file but they only contain the source code which is contained in [here](https://github.com/datarootsio/skeleton-pyspark/tree/main/src). It also contains a METADATA file which tells it about the dependencies. If Databricks/Synapse is smart enough to infer and install these dependencies based on METADATA, then great. If not, I believe, the dependencies which are packaged in the `make pack_req` should be sent to the Databricks/Synapse

**Comment by fbraza** [comment]
Thank you for your feedback @vikramaditya91. Concerning your point, if you install the package from the wheel it does also install the dependencies. You can make the test locally.  Add boto3 and numpy, build the `.whl` and then `pip install your_whell.whl` and you can see that it install all dependencies. So this should not be a problem once you are in Databricks.

**Comment by vikramaditya91** [comment]
@fbraza Indeed, I just confirmed this with a Databricks notebook. When a `.whl` package is uploaded, it installs all the dependencies based on the METADATA file. (`pip install` uses the METADATA file too). So this PR is good for me.

Confirming it locally would not have been sufficient, because you want to emulate how Databricks would install the package. If it simply did a `unzip` on `.whl` file, it would have been insufficient.

**Comment by fbraza** [comment]
Good ! Yes I was trying on databricks community but could not have a cluster spinning up ^_^. 

Thx for having made the test yourself !

File: Makefile [hunk 1]
```diff
@@ -26,6 +26,9 @@ coverage: ## create coverage report
 	poetry run coverage run --source=src -m pytest
 	poetry run coverage xml
 
+build: ## create the wheel
```
Comment (vikramaditya91) [inline Makefile:29]: @fbraza Could you please move this to around line 46-67. This section is about testing and linting. And there are things related to packaging and submitting in there

**Review by vikramaditya91 — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (fbraza) [reply Makefile]: Done @vikramaditya91

**Pull request merged**